encoding/protobuf_test.go

```diff
@@ -0,0 +1,167 @@
+package encoding
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/yarpc/yab/protobuf"
+	"github.com/yarpc/yab/transport"
+)
+
+func TestNewProtobuf(t *testing.T) {
+	tests := []struct {
+		desc   string
+		method string
+		errMsg string
+	}{
+		{
+			desc:   "simple",
+			method: "Bar/Baz",
+		},
+		{
+			desc:   "no method",
+			method: "Bar",
+			errMsg: `service "Bar" does not include a method named ""`,
+		},
+		{
+			desc:   "invalid method",
+			method: "Bar/Baz/Foo",
+			errMsg: `invalid proto method "Bar/Baz/Foo", expected form package.Service/Method`,
+		},
+		{
+			desc:   "service not found",
+			method: "Baq/Foo",
+			errMsg: `failed to query for service for symbol "Baq"`,
+		},
+		{
+			desc:   "service not found but symbol is",
+			method: "Foo/Foo",
+			errMsg: `target server does not expose service "Foo"`,
+		},
+	}
+	source, err := protobuf.NewDescriptorProviderFileDescriptorSetBins("../testdata/protobuf/simple/simple.proto.bin")
+	require.Nil(t, err)
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			serializer, err := NewProtobuf(tt.method, source)
+			if tt.errMsg == "" {
+				require.NoError(t, err)
+				require.NotNil(t, serializer)
+				assert.Equal(t, Protobuf, serializer.Encoding(), "Encoding mismatch")
+			} else {
+				require.Error(t, err, "%v", tt.desc)
+				require.Nil(t, serializer, "%v: Error cases should not return a serializer", tt.desc)
+				assert.Contains(t, err.Error(), tt.errMsg, "%v: invalid error", tt.desc)
+			}
+		})
+	}
+}
+
+func TestProtobufRequest(t *testing.T) {
+	tests := []struct {
+		desc   string
+		bsIn   []byte
```

Comment:
the test cases are using strings for input, so make this `input string` and then convert to `[]byte` as part of the test execution logic